packages/react-native-bpk-component-button/src/BpkButton.android.js

```diff
@@ -19,108 +19,223 @@
 /* @flow */
 
 import React from 'react';
-import { withTheme } from 'react-native-bpk-theming';
 import BpkIcon from 'react-native-bpk-component-icon';
-import BpkText, { WEIGHT_STYLES } from 'react-native-bpk-component-text';
-import BpkTouchableNativeFeedback from 'react-native-bpk-component-touchable-native-feedback';
-import { Platform, View } from 'react-native';
-
 import {
-  isTypeThemeable,
-  themeAttributesSupplied,
-  getStyleForElement,
-  getAndroidBackgroundColour,
-  textStyle,
-  iconStyle,
-} from './utils';
+  getThemeAttributes,
+  withTheme,
+  type Theme,
+} from 'react-native-bpk-theming';
+import {
+  colorBlue500,
+  colorGray100,
+  colorGray300,
+  colorGreen500,
+  colorPink500,
+  colorRed500,
+  colorWhite,
+} from 'bpk-tokens/tokens/base.react.native';
+
+import BpkFlatButton from './BpkFlatButton.android';
+import BpkBorderedButton from './BpkBorderedButton.android';
+import BpkButtonInner from './BpkButtonInner';
 import {
+  type ButtonType,
   type CommonProps,
   commonPropTypes,
   commonDefaultProps,
   BUTTON_TYPES,
   ICON_ALIGNMENTS,
+  REQUIRED_THEME_ATTRIBUTES,
 } from './common-types';
 
-export type Props = {
-  ...$Exact<CommonProps>,
+export type Props = CommonProps;
+
+const backgroundColors = {
+  disabled: colorGray100,
+  [BUTTON_TYPES.primary]: colorGreen500,
+  [BUTTON_TYPES.featured]: colorPink500,
+  [BUTTON_TYPES.secondary]: colorWhite,
+  [BUTTON_TYPES.destructive]: colorWhite,
+};
+
+const borderColors = {
+  disabled: colorGray100,
+  [BUTTON_TYPES.secondary]: colorGray100,
+  [BUTTON_TYPES.destructive]: colorGray100,
+};
+
+const textColors = {
+  [BUTTON_TYPES.primary]: colorWhite,
+  [BUTTON_TYPES.featured]: colorWhite,
+  [BUTTON_TYPES.secondary]: colorBlue500,
+  [BUTTON_TYPES.destructive]: colorRed500,
+  disabled: colorGray300,
+};
+
+const getButtonComponent = (type: ButtonType) => {
+  if (type === BUTTON_TYPES.primary || type === BUTTON_TYPES.featured) {
+    return BpkFlatButton;
+  }
+  return BpkBorderedButton;
+};
+
+const getButtonColors = (
+  type: ButtonType,
+  themeAttributes: ?Theme,
+  disabled: boolean,
+) => {
+  if (type === BUTTON_TYPES.primary) {
```

Comment:
This method could use a lot more vertical space to separate the different cases I think.

Reply:
👍 Will do. I was concerned that this huge if..else would be hard to read tbh, but it felt better than over-optimising like the old implementation.